src/ray/raylet/scheduling/scheduling_policy.cc

```diff
@@ -138,6 +130,9 @@ int64_t SchedulingPolicy::HybridPolicyWithFilter(
     }
 
     if (update_best_node) {
+      if (spread_threshold == 0) {
+        spread_scheduling_next_index_ = ((round_index + 1) % round.size());
```

Comment:
nit, but why don't we just set
```
size_t round_index = scheduling_next_index_
```

and change this code to 
```
      if (spread_threshold == 0) {
        // SPREAD
        scheduling_next_index_ = ((round_index + 1) % round.size());
      } else {
         // PACK
         scheduling_next_index_ = 0;
     }
```

Reply:
As @clarkzinzow mentioned, if we do `scheduling_next_index_ = 0;`, a pack task will affect the round robin of the spread task.

Reply:
I think a lot of complexity comes from the mixture of pack and spread policy in the same function. wondering if it's easier if we separate those two.

Reply:
They actually have almost the same implementation: the only difference is where we start the scan.